package/DataTableResizableHeaderKnob.tsx

```diff
@@ -0,0 +1,85 @@
+import { rem } from '@mantine/core';
+import { MutableRefObject, useRef, useState } from 'react';
+import { useDataTableDragToggleColumnsContext } from './DataTableDragToggleColumns.context';
+
+type DataTableResizableHeaderCellProps = {
+  accessor: string;
+  columnRef: MutableRefObject<HTMLTableCellElement | null>;
+};
+
+export const DataTableResizableHeaderKnob = (props: DataTableResizableHeaderCellProps) => {
+  const { accessor, columnRef } = props;
+
+  const dragRef = useRef<HTMLTableCellElement>(null);
+
+  const [deltaX, setDeltaX] = useState<number>(0);
+
+  const { setColumnWidth } = useDataTableDragToggleColumnsContext();
+
+  const handleDragStart = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    document.addEventListener('mousemove', handleMouseMove);
+    document.addEventListener('mouseup', handleMouseUp);
+
+    document.body.style.cursor = 'col-resize';
+  };
+
+  const handleMouseMove = (event: MouseEvent) => {
+    if (!columnRef.current) return;
+
+    const delta = event.clientX - columnRef.current.getBoundingClientRect().right;
+
+    const width = columnRef.current.getBoundingClientRect().width + delta;
+
+    const widthString = `${width}px`;
+
+    columnRef.current.style.width = widthString;
+
+    // we have to set (store) the width in the context  and in the
+    // local storage, otherwise the resizing won't work for small sizes
+    setColumnWidth(accessor, columnRef.current.style.width as string);
+
+    setDeltaX(-delta);
+  };
+
+  const handleMouseUp = () => {
+    if (!columnRef.current) return;
+
+    document.removeEventListener('mousemove', handleMouseMove);
+    document.removeEventListener('mouseup', handleMouseUp);
+
+    document.body.style.cursor = 'initial';
+
+    setColumnWidth(accessor, columnRef.current.style.width as string);
+
+    setDeltaX(0);
+  };
+
+  /**
+   * Reset the column width to the default value
+   */
+  const handleDoubleClick = () => {
+    if (!columnRef.current) return;
+
+    columnRef.current.style.maxWidth = 'initial';
+    columnRef.current.style.minWidth = 'initial';
+    columnRef.current.style.width = 'initial';
+
+    setColumnWidth(accessor, 'initial');
+  };
+
+  return (
+    <div
+      ref={dragRef}
+      onMouseDown={handleDragStart}
```

Comment:
This is needed to stop the click event from propagating to the column header; otherwise, column sorting (if presend) will be triggered when clicking the knowb.

Reply:
yep, I saw now. Fixing 👍

Reply:
No need, already did it